src/inference/gibbs.jl

```diff
@@ -29,17 +29,33 @@ methods like Particle Gibbs. You can increase the effectiveness of particle samp
 more particles in the particle sampler.
 """
 mutable struct Gibbs{A} <: InferenceAlgorithm
-    n_iters   ::  Int     # number of Gibbs iterations
     algs      ::  A   # component sampling algorithms
-    thin      ::  Bool    # if thinning to output only after a whole Gibbs sweep
+    function Gibbs(algs...)
+        algs = algs isa Tuple ? algs : (algs, )
+        return new{typeof(algs)}(algs)
+    end
+end
+
+alg_str(::Sampler{<:Gibbs}) = "Gibbs"
+transition_type(spl::Sampler{<:Gibbs}) = Transition
+
+mutable struct GibbsState{T<:NamedTuple} <: AbstractSamplerState
+    vi::TypedVarInfo
+    samplers::Array{Sampler}
```

Comment:
Can we make this a tuple? Do the length or types change after initial construction? If tuple, then it should be a type parameter `<: Tuple{Vararg{Sampler}}`.

Reply:
Yeah, we can make it a tuple. Good idea. I've added it in.